backend/amd64/simd_selection.ml

```diff
@@ -463,3 +463,347 @@ let () =
   Location.register_error_of_exn (function
     | Error err -> Some (Location.error_of_printer_file report_error err)
     | _ -> None)
+
+type width =
+  | W8
+  | W16
+  | W32
+  | W64
+
+type register =
+  | New of int
+  | Argument of int
+  | Result of int
+  | Original of int
+
+type vectorized_instruction =
+  { operation : Cfg.operation;
+    arguments : register array;
+    results : register array
+  }
+
+let vector_width = 128
+
+let vectorize_operation width (cfg_ops : Cfg.operation list) :
+    vectorized_instruction list option =
+  (* Assumes cfg_ops are isomorphic and can be vectorized *)
+  let length = List.length cfg_ops in
+  assert (length * width = vector_width);
+  let width_type =
+    match width with
+    | 64 -> W64
+    | 32 -> W32
+    | 16 -> W16
+    | 8 -> W8
+    | _ -> assert false
+  in
+  let make_default arg_count res_count operation =
+    Some
+      [ { operation;
+          arguments = Array.init arg_count (fun i -> Argument i);
+          results = Array.init res_count (fun i -> Result i)
+        } ]
+  in
+  let create_const_vec consts =
+    let rec get_list list n =
+      if n = 0
+      then [], list
+      else
+        match list with
+        | [] -> assert false
+        | hd :: tl ->
+          let list1, list2 = get_list tl (n - 1) in
+          hd :: list1, list2
```

Comment:
(Isn't it `Misc.List.split_at`?)

Reply:
I can't seem to find that

Reply:
you mean Misc.split_at?

Reply:
I seem to not be able to call that either

Reply:
ah it's Misc.Stdlib.List.split_at